### PR TITLE
Default to EGL initialization if no options are provided and OS is at least Windows 8

### LINKED ISSUE #5281: Windows EGL initialization has inconsistent default
The following line is a bit too clever I believe:
https://github.com/AvaloniaUI/Avalonia/blob/25fe106c235ebb5c6260af5a13d6a79b633c5b82/src/Windows/Avalonia.Win32/Win32GlManager.cs#L24

If `opts` is null, the default is false, since the negation will be applied before the null-coalescing operator.
If `opts` is not null, but `AllowEglInitialization` is null, it defaults to true.

I'll submit a fix PR if you confirm the **intended** functionality is to default to EGL (only on Windows 8 or later).

### PR DESCRIPTION
**Pull request by shartte**
## What does the pull request do?
#5281 

## What is the current behavior?
If options are not provided, EGL initialization defaults to false, otherwise it defaults to true.

## What is the updated/expected behavior with this PR?
EGL initialization will default to true if not specified otherwise, and OS is newer than Windows 7.

## Fixed issues
Fixes #5281

### CODE REVIEW

File: src/Windows/Avalonia.Win32/Win32GlManager.cs [hunk 1]
```diff
@@ -20,13 +22,11 @@ public static void Initialize()
...
 
-                if (opts?.AllowEglInitialization == true ||
-                    ((!opts?.AllowEglInitialization.HasValue ?? false) &&
-                     Win32Platform.WindowsVersion > new Version(6, 1)))
+                if (opts?.AllowEglInitialization ?? Win32Platform.WindowsVersion > Windows7)
                 {
                     var egl = EglPlatformOpenGlInterface.TryCreate(() => new AngleWin32EglDisplay());
 
-                    if (egl is { } &&
+                    if (egl != null &&
```
Comment (rstm-sf) [inline src/Windows/Avalonia.Win32/Win32GlManager.cs:29]: C#9 introduced the `is not` operator. So it seems that this unnecessary change will be in this pr :)

Comment (shartte) [reply src/Windows/Avalonia.Win32/Win32GlManager.cs:29]: This seemed like fancyness for fancess sake. It's a not-null check....

Comment (rstm-sf) [reply src/Windows/Avalonia.Win32/Win32GlManager.cs:29]: However, this is how we can write https://dotnetfiddle.net/0AgVfr

Using this hack to avoid writing  `!(string.Empty is null)`

**Review by maxkatz6 — APPROVED** [review]
LGTM
